spec/jobs/risc_delivery_job_spec.rb

```diff
@@ -91,25 +112,51 @@
       end
     end
 
-    context 'Errno::ECONNREFUSED error' do
+    it_behaves_like 'retries and logs event when attempts exhausted', Faraday::SSLError
+    it_behaves_like 'retries and logs event when attempts exhausted', Faraday::ConnectionFailed
+
+    context 'when the job fails due to an Errno::ECONNREFUSED error' do
       before do
+        allow_any_instance_of(described_class).to receive(:analytics).and_return(job_analytics)
         # stub_request().to_raise wraps this in Faraday::ConnectionFailed, but
         # in actual usage, the original error is unwrapped
-        expect(job.faraday).to receive(:post).and_raise(Errno::ECONNREFUSED)
+        @connection = instance_double(Faraday::Connection)
+        allow(@connection).to receive(:post).and_raise(Errno::ECONNREFUSED)
+        allow(Faraday).to receive(:new).and_return(@connection)
       end
 
-      it 'raises and retries via ActiveJob' do
-        expect { perform }.to raise_error(Errno::ECONNREFUSED)
-      end
-
-      context 'it has already failed twice' do
-        before do
-          allow(job).to receive(:executions).and_return 2
+      context 'when the job fails for the 1st time' do
+        it 'raises and retries via ActiveJob' do
+          expect do
+            RiscDeliveryJob.new.perform(
```

Comment:
`expect { perform }.to raise_error(Errno::ECONNREFUSED)`

Reply:
ditto